jaas/src/main/java/com/griddynamics/jagger/jaas/rest/TestEnvironmentRestController.java

```diff
@@ -102,13 +103,21 @@ public TestEnvironmentRestController(TestEnvironmentService testEnvironmentServi
 
         if (!testEnvService.existsWithSessionId(envId, sessionId))
             throw new TestEnvironmentSessionNotFoundException(envId, sessionId);
+        validateTestEnv(testEnv);
+        TestEnvironmentEntity oldEnv = testEnvService.read(envId);
 
         testEnv.setEnvironmentId(envId);
         TestEnvironmentEntity updated = testEnvService.update(testEnv);
-        response.addCookie(getSessionCookie(updated));
         if (updated.getStatus() == PENDING) {
-            getTestSuiteNameToExecute(updated).ifPresent(testSuiteName -> setNextConfigToExecuteHeader(response, testSuiteName));
+            if (oldEnv.getStatus() == RUNNING)
+                jobExecutionService.finishExecution(envId, oldEnv.getRunningTestSuite().getTestSuiteId());
```

Comment:
Because getTestSuiteNameToExecute finds all pending jobs, and for finishing we need running ones. 
Since one job cannot have multiple simultaneous executions, there's only one job with status RUNNING for any pair env-testsuite.